#### Requirements:
Determine if the shade of lipstick and the price category are related.

- Create a contingency table
- Test for the assumption of 5 per cell in the expected contingency table
- Compute an independent Chi-Square

#### Import Packages

In [1]:
import pandas as pd
from scipy import stats

#### Import Data and view

In [2]:
LeadLipstick = pd.read_csv('C:/Users/desja/python_course/datasets/LeadLipstick.csv')
LeadLipstick.head()

JRC_code purchCntry prodCntry    Pb  sdPb shade prodType  priceCatgry
0     C135         NL        NL  3.75  0.24   Red       LP            2
1      C18         FI        FI  2.29  0.07   Red       LP            2
2      C20         FI        IT  1.27  0.06   Red       LP            2
3     C164         DE        FR  1.21  0.06   Red       LP            2
4      C71         MT        UK  0.85  0.04   Red       LP            2

#### Create a contingency table

In [3]:
LeadLipstick_crosstab = pd.crosstab(LeadLipstick['shade'], LeadLipstick['priceCatgry'])
LeadLipstick_crosstab

priceCatgry   1   2   3
shade                  
Brown        20  30  10
Pink         20  49  12
Purple        8  23   6
Red           5  33   7

#### Run the Independent Chi-Square

In [4]:
stats.chi2_contingency(LeadLipstick_crosstab)

Chi2ContingencyResult(statistic=7.860569553614045, pvalue=0.2484973879479863, dof=6, expected_freq=array([[14.26008969, 36.32286996,  9.41704036],
       [19.25112108, 49.03587444, 12.71300448],
       [ 8.79372197, 22.39910314,  5.80717489],
       [10.69506726, 27.24215247,  7.06278027]]))

#### Results 
- The first value is the Chi-Square statistic=7.860569553614045 
- The pvalue=0.2484973879479863 is above 0.05
  This means that there is no significant relationship between the shade of lipstick and the price category

- Since all of these values are over 5, this means that the assumption has been met